### PR TITLE
[#3] [UI] As a user, I can switch between light/dark mode base on System settings

### LINKED ISSUE #3: [UI] As a user, I can switch between light/dark mode base on System settings
## Why

For better UX, the application support users who are prefer to use device in low-light environment.

## Acceptance Criteria

Switch the background color base on System settings:
- Light -> `#D0DEDD`
- Dark -> `#141B29` 

## Design

image

## Resources

https://www.figma.com/file/HioYp71LllKVy9qaiLauKn/Crypto-App-(Community)?node-id=3%3A166

### PR DESCRIPTION
**Pull request by luongvo**
Resolves https://github.com/nimblehq/jetpack-compose-crypto/issues/3

## What happened 👀

- Set dark/light background and status bar colors for `Compose`.
  - use `accompanist-systemuicontroller` to `systemUiController.setSystemBarsColor`.
- Set dark/light background and status bar colors for `base wrapper activity`.
  - `material` 1.6.0 requires API 31 and Gradle 7.
- Correct app name and remove unused layout.

## Insight 📝

N/A

## Proof Of Work 📹

https://user-images.githubusercontent.com/16315358/188708295-95fabc3a-2dea-4f1d-ae60-733e70b349b9.mp4

https://user-images.githubusercontent.com/16315358/188708333-35b2a664-ffe9-48cd-82ed-2e9a7caccc7d.mp4

### CODE REVIEW

**Comment by Wadeewee** [comment]
@luongvo Could you please rebase branch with develop 🙏

**Comment by luongvo** [comment]
> @luongvo Could you please rebase branch with develop 🙏

Done @Wadeewee

File: app/src/staging/res/values/strings.xml [hunk 1]
```diff
@@ -1,3 +1,3 @@
 <resources>
-    <string name="app_name">Cryto Compose - Staging</string>
+    <string name="app_name">Crypto Compose - Staging</string>
```
Comment (hoangnguyen92dn) [inline app/src/staging/res/values/strings.xml:2]: Highly appreciate your eagle eyes 🦅 👀

File: app/src/main/java/co/nimblehq/compose/crypto/ui/theme/Color.kt [hunk 2]
```diff
@@ -5,7 +5,7 @@ import androidx.compose.ui.graphics.Color
 @Suppress("MagicNumber")
 object Color {
     val BlueFreeSpeech = Color(0xFF3F51B5)
-    val White = Color.White
+    val AlmostWhite = Color(0xFFF8F8F8)
```
Comment (hoangnguyen92dn) [inline app/src/main/java/co/nimblehq/compose/crypto/ui/theme/Color.kt]: `AlmostWhite` :trollface: 

image

File: app/build.gradle.kts [hunk 3]
```diff
@@ -137,11 +137,14 @@ dependencies {
     implementation("androidx.compose.ui:ui-tooling:${Versions.COMPOSE_VERSION}")
     implementation("androidx.compose.foundation:foundation:${Versions.COMPOSE_VERSION}")
     implementation("androidx.compose.material:material:${Versions.COMPOSE_VERSION}")
+    implementation("com.google.accompanist:accompanist-systemuicontroller:${Versions.COMPOSE_ACCOMPANIST_VERSION}")
```
Comment (Wadeewee) [inline app/build.gradle.kts]: We already have `androidx.compose.foundation:foundation` why do we have to add this 🙏

Comment (luongvo) [reply app/build.gradle.kts]: @Wadeewee we need to use `rememberSystemUiController` 🤔

File: app/src/main/java/co/nimblehq/compose/crypto/ui/theme/Theme.kt [hunk 4]
```diff
@@ -33,6 +33,16 @@ fun ComposeTheme(
         Palette.ComposeLightPalette
     }
 
+    val systemUiController = rememberSystemUiController()
+    DisposableEffect(systemUiController, darkTheme) {
```
Comment (Wadeewee) [inline app/src/main/java/co/nimblehq/compose/crypto/ui/theme/Theme.kt]: @luongvo Regarding this  [comment](https://github.com/nimblehq/jetpack-compose-crypto/pull/30#discussion_r965736522) I don't think we need `rememberSystemUiController` as we can use `systemBarsPadding()` from `androidx.compose.foundation`, Could you please following these steps.
1. Comment all changes in `Theme.kt`.
2. In `MainActivity` add `WindowCompat.setDecorFitsSystemWindows(window, false)` inside `onCreate` method and put it above `setCotent` scope.
3. Upgrade `COMPOSE_VERSION` to `1.2.0-rc01` (You need to upgrade `ANDROID_COMPILE_SDK_VERSION` -> `32` and `KOTLIN_VERSION` to `1.6.21`)
4. In `HomeScreen` In line `29` add `modifier = Modifier.fillMaxSize().systemBarsPadding()`
5. In `style.xml` comment these item:
   
Screen Shot 2565-09-08 at 17 20 06
6. Run and test

Comment (luongvo) [reply app/src/main/java/co/nimblehq/compose/crypto/ui/theme/Theme.kt]: @Wadeewee Okay, 
- using `WindowCompat.setDecorFitsSystemWindows(window, false)` and `Modifier.fillMaxSize().systemBarsPadding()` works for me ✅ 
- We will need to use `rememberSystemUiController` if we need to switch between `immersive` and `non-immersive` screen content in each screen later.
- we still need to keep `<item name="android:windowBackground">@color/colorBackground</item>` to set the correct background at app starting up.
- we still need to keep `<item name="android:navigationBarColor">@android:color/transparent</item>` to make the full immersive effect for both top status bar and bottom navigation bar too.

Comment (luongvo) [reply app/src/main/java/co/nimblehq/compose/crypto/ui/theme/Theme.kt]: @Wadeewee Fixed in https://github.com/nimblehq/jetpack-compose-crypto/pull/30/commits/0c761c0af38631e1bdd7babc5aca8a34c1482398

File: app/src/main/java/co/nimblehq/compose/crypto/ui/theme/Theme.kt [hunk 5]
```diff
@@ -1,19 +1,17 @@
...
 import androidx.compose.runtime.Composable
-import co.nimblehq.compose.crypto.ui.theme.Color.Cultured
 import co.nimblehq.compose.crypto.ui.theme.Color.BlueFreeSpeech
 import co.nimblehq.compose.crypto.ui.theme.Color.DarkJungleGreen
+import co.nimblehq.compose.crypto.ui.theme.Color.Guyabano
 
 @Suppress("MatchingDeclarationName")
 object Palette {
     val ComposeLightPalette = lightColors(
         primary = BlueFreeSpeech,
```
Comment (Wadeewee) [inline app/src/main/java/co/nimblehq/compose/crypto/ui/theme/Theme.kt]: I think we can remove this, Also for the color `BlueFreeSpeech`

File: app/src/main/res/values/styles.xml [hunk 6]
```diff
@@ -1,11 +1,11 @@
 <resources>
 
     <!-- Base application theme. -->
-    <style name="AppTheme" parent="Theme.AppCompat.Light.NoActionBar">
-        <!-- Customize your theme here. -->
-        <item name="colorPrimary">@color/colorPrimary</item>
-        <item name="colorPrimaryDark">@color/colorPrimaryDark</item>
-        <item name="colorAccent">@color/colorAccent</item>
+    <style name="AppTheme" parent="Theme.Material3.DayNight.NoActionBar">
```
Comment (Wadeewee) [inline app/src/main/res/values/styles.xml]: @luongvo I think we can use this theme `android:Theme.Material.Light.NoActionBar` So, we don't add `com.google.android.material:material`
```suggestion
    <style name="AppTheme" parent="android:Theme.Material.Light.NoActionBar">
```

File: app/src/main/res/values/styles.xml [hunk 7]
```diff
@@ -1,11 +1,11 @@
...
+        <item name="android:statusBarColor">@android:color/transparent</item>
+        <item name="android:navigationBarColor">@android:color/transparent</item>
+        <item name="android:windowBackground">@color/colorBackground</item>
+        <item name="android:windowLightStatusBar">@bool/windowLightStatusBar</item>
```
Comment (Wadeewee) [inline app/src/main/res/values/styles.xml:8]: Regarding [this](https://github.com/nimblehq/jetpack-compose-crypto/pull/30/files#r966613011) I think we can remove these 👇 
```suggestion
```

**Review by Wadeewee — COMMENTED** [review]
I'm done my review for the rest LGTM 🚀

File: app/src/main/java/co/nimblehq/compose/crypto/ui/theme/Theme.kt (hunk 4 above)
Comment (Wadeewee) [reply app/src/main/java/co/nimblehq/compose/crypto/ui/theme/Theme.kt]: >- We will need to use rememberSystemUiController if we need to switch between immersive and non-immersive >screen content in each screen later.
>- we still need to keep <item name="android:windowBackground">@color/colorBackground</item> to set the >correct background at app starting up.
>- we still need to keep <item name="android:navigationBarColor">@android:color/transparent</item> to make the >full immersive effect for both top status bar and bottom navigation bar too.

@luongvo I'm not sure about these 🙏 So Let's keep it then, I will recheck after this PR was merged and test on this [PR](https://github.com/nimblehq/jetpack-compose-crypto/pull/28) again, If we need it🚀

File: app/src/main/res/values/styles.xml (hunk 6 above)
Comment (luongvo) [reply app/src/main/res/values/styles.xml]: @Wadeewee `Theme.Material3.DayNight.NoActionBar` does not support dark background properly before having compose background rendering.

File: app/src/main/res/values/styles.xml (hunk 7 above)
Comment (luongvo) [reply app/src/main/res/values/styles.xml:8]: @Wadeewee okay, so let's keep it then https://github.com/nimblehq/jetpack-compose-crypto/pull/30#discussion_r966616541

File: app/src/main/res/values/styles.xml (hunk 6 above)
Comment (luongvo) [reply app/src/main/res/values/styles.xml]: @Wadeewee After checked again, we can remove it and use the current style.

**Review by Wadeewee — APPROVED** [review]

**Review by hoangnguyen92dn — APPROVED** [review]
🚀

**Pull request merged**